routing/localchans/manager.go

```diff
@@ -75,17 +81,37 @@ func (r *Manager) UpdatePolicy(newSchema routing.ChannelPolicy,
 
 		// If we have a channel filter, and this channel isn't a part
 		// of it, then we'll skip it.
-		_, ok := chansToUpdate[info.ChannelPoint]
+		_, ok := unprocessedChans[info.ChannelPoint]
 		if !ok && haveChanFilter {
 			return nil
 		}
 
+		// Mark this channel as found by removing it. unprocessedChans
+		// will be used to report invalid channels later on.
+		delete(unprocessedChans, info.ChannelPoint)
+
 		// Apply the new policy to the edge.
 		err := r.updateEdge(tx, info.ChannelPoint, edge, newSchema)
 		if err != nil {
-			log.Warnf("Cannot update policy for %v: %v\n",
-				info.ChannelPoint, err,
-			)
+			failedUpdates = append(failedUpdates,
+				makeFailureItem(info.ChannelPoint,
+					lnrpc.UpdateFailure_UPDATE_FAILURE_INVALID_PARAMETER,
+					err.Error(),
+				))
+
+			return nil
+		}
+
+		// Check whether the channel is inactive, if so, we add an item
+		// to the failed policy update list.
+		channelID := lnwire.NewChanIDFromOutPoint(&info.ChannelPoint)
+		if !r.IsChannelActive(channelID) {
```

Comment:
Why should we care if a channel is inactive or not? The link existing doesn't mean that we can't update the channel on the gossip layer. 

This change would likely also negatively interact with an attempt to disable a link (manually) when a channle is inactive.

Reply:
> Edit: As is, the policy update command silently fails to update a channel policy if it the channel is inactive. The latest update fixes this by skipping the policy update and returns a policy update report.

Ah I see this now, can you point to where it would fail before?

Reply:
@Roasbeef I ran some tests and it looks like the policy update command **does not** silently fails when a channel is inactive on the latest master build. I tried to track down where this issue would have occurred (as described in the comment section of issue #4846), all the way back to 0.12.0-beta, but could not figure it out. 

If you want me to include inactive channels in the policy update process I'll update the pr right away.

Reply:
I ran some tests and it looks like the policy update command works as it should on inactive channels. I tried to track down where this issue would have occurred (as described in the comment section of issue #4846), all the way back to 0.12.0-beta, but could not figure it out. I didn't realize the gossip layer would take care of it. My bad.
Anyway, the latest commit leaves inactive channels alone and they will be treated the same way as in master.

Reply:
@Roasbeef Don't have any answers right now. I'll try to address the issue in a separate pr.